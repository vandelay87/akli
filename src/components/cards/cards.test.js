import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import {
  card,
  cardWithSubtitle,
  cardsWithImage,
  cardWithLink,
} from './cards.stories'

describe('cards.js', () => {
  it('renders component', () => {
    const cardTree = renderer.create(card()).toJSON()
    const cardWithSubtitleTree = renderer.create(cardWithSubtitle()).toJSON()
    const cardsWithImageTree = renderer.create(cardsWithImage()).toJSON()
    const cardWithLinkTree = renderer.create(cardWithLink()).toJSON()

    expect(cardTree).toMatchSnapshot()
    expect(cardWithSubtitleTree).toMatchSnapshot()
    expect(cardsWithImageTree).toMatchSnapshot()
    expect(cardWithLinkTree).toMatchSnapshot()
  })

  it('opens url when clicked', () => {
    window.open = jest.fn()
    const wrapper = mount(cardWithLink())

    wrapper.find('MaterialButton').simulate('click')
    expect(window.open).toHaveBeenCalledTimes(1)
  })
})
