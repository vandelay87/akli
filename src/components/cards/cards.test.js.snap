// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cards.js renders component 1`] = `
<div
  className="sc-EHOje gAclcb"
>
  <article
    className="sc-bZQynM eKUKWx"
  >
    <div
      className="sc-gzVnrw jIssoY"
    >
      <h2
        className="sc-bdVaJa kAaGL"
        size={2}
      >
        Lorem Ipsum
      </h2>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
      </p>
    </div>
  </article>
</div>
`;

exports[`cards.js renders component 2`] = `
<div
  className="sc-EHOje gAclcb"
>
  <article
    className="sc-bZQynM eKUKWx"
  >
    <div
      className="sc-gzVnrw jIssoY"
    >
      <h2
        className="sc-bdVaJa kAaGL"
        size={2}
      >
        Lorem Ipsum
      </h2>
      <h3
        className="sc-bdVaJa bOQwgh"
        size={3}
      >
        Duis aute irure
      </h3>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
      </p>
    </div>
  </article>
</div>
`;

exports[`cards.js renders component 3`] = `
<div
  className="sc-EHOje gAclcb"
>
  <article
    className="sc-bZQynM eKUKWx"
  >
    <figure
      className="sc-bwzfXH cmJVxu"
    >
      <img
        src="https://via.placeholder.com/400x200"
      />
      
    </figure>
    <div
      className="sc-gzVnrw jIssoY"
    >
      <h2
        className="sc-bdVaJa kAaGL"
        size={2}
      >
        Facebook
      </h2>
      <h3
        className="sc-bdVaJa bOQwgh"
        size={3}
      >
        email@facebook.com
      </h3>
      <p>
        Connect with friends, family and other people you know. Share photos and videos, send messages and get updates.
      </p>
    </div>
  </article>
  <article
    className="sc-bZQynM eKUKWx"
  >
    <figure
      className="sc-bwzfXH cmJVxu"
    >
      <img
        src="https://via.placeholder.com/400x200"
      />
      
    </figure>
    <div
      className="sc-gzVnrw jIssoY"
    >
      <h2
        className="sc-bdVaJa kAaGL"
        size={2}
      >
        Twitter
      </h2>
      <h3
        className="sc-bdVaJa bOQwgh"
        size={3}
      >
        email@twitter.com
      </h3>
      <p>
        See what is happening in the world right now. Join Twitter today. Sign up Log In.
      </p>
    </div>
  </article>
  <article
    className="sc-bZQynM eKUKWx"
  >
    <figure
      className="sc-bwzfXH cmJVxu"
    >
      <img
        src="https://via.placeholder.com/400x200"
      />
      
    </figure>
    <div
      className="sc-gzVnrw jIssoY"
    >
      <h2
        className="sc-bdVaJa kAaGL"
        size={2}
      >
        LinkedIn
      </h2>
      <h3
        className="sc-bdVaJa bOQwgh"
        size={3}
      >
        email@linkedin.com
      </h3>
      <p>
        Manage your professional identity. Build and engage with your professional network. Access knowledge, insights and opportunities.
      </p>
    </div>
  </article>
</div>
`;

exports[`cards.js renders component 4`] = `
<div
  className="sc-EHOje gAclcb"
>
  <article
    className="sc-bZQynM eKUKWx"
  >
    <div
      className="sc-gzVnrw jIssoY"
    >
      <h2
        className="sc-bdVaJa kAaGL"
        size={2}
      >
        Lorem Ipsum
      </h2>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
      </p>
    </div>
    <button
      className="sc-ifAKCX bRkpKd"
      onClick={[Function]}
      role="button"
    >
      View
      <div
        className="sc-bxivhb kyOqOf"
        color="#1976d2"
        onMouseDown={[Function]}
      />
    </button>
  </article>
</div>
`;
